src/back/spv/writer.rs

```diff
@@ -688,11 +691,22 @@ impl Writer {
             next_id
         };
 
+        let debug_info = (|| {
+            if !context.writer.flags.contains(WriterFlags::DEBUG) {
+                return None;
+            }
+            Some(DebugInfo {
+                source_code: ir_module.source_code.as_ref()?,
+                source_file_id: context.writer.source_file_id?,
+            })
+        })();
```

Comment:
We can use [`then_some`](https://doc.rust-lang.org/std/primitive.bool.html#method.then_some) here.